<!DOCTYPE html>

<html lang="en">
<head>
<meta charset="utf-8"/>
<meta content="IE=edge" http-equiv="X-UA-Compatible"/>
<meta content="width=device-width, initial-scale=1.0" name="viewport"/>
<meta content="None" name="description"/>
<link href="img/favicon.ico" rel="shortcut icon"/>
<title>EWAS-fusion</title>
<link href="css/bootstrap.min.css" rel="stylesheet"/>
<link href="css/font-awesome.min.css" rel="stylesheet"/>
<link href="css/base.css" rel="stylesheet"/>
<link href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css" rel="stylesheet"/>
<script defer="" src="js/jquery-1.10.2.min.js"></script>
<script defer="" src="js/bootstrap.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/languages/Bash.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/languages/HTML.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/languages/Java.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/languages/JavaScript.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/languages/Perl.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/languages/Python.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/languages/R.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/languages/rust.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/languages/yaml.min.js"></script>
<script>hljs.initHighlightingOnLoad();</script>
</head>
<body class="homepage">
<div class="navbar fixed-top navbar-expand-lg navbar-dark bg-dark">
<div class="container">
<a class="navbar-brand" href=".">EWAS-fusion</a>
<!-- Expanded navigation -->
<div class="navbar-collapse collapse" id="navbar-collapse">
<ul class="nav navbar-nav ml-auto">
<li class="nav-item">
<a class="nav-link" data-target="#mkdocs_search_modal" data-toggle="modal" href="#">
<i class="fa fa-search"></i> Search
                            </a>
</li>
</ul>
</div>
</div>
</div>
<div class="container">
<div class="row">
<div class="col-md-3"><div class="navbar-light navbar-expand-md bs-sidebar hidden-print affix" role="complementary">
<div class="navbar-header">
<button class="navbar-toggler collapsed" data-target="#toc-collapse" data-toggle="collapse" title="Table of Contents" type="button">
<span class="fa fa-angle-down"></span>
</button>
</div>
<div class="navbar-collapse collapse card bg-secondary" id="toc-collapse">
<ul class="nav flex-column">
<li class="nav-item" data-level="2"><a class="nav-link" href="#introduction">INTRODUCTION</a>
<ul class="nav flex-column">
</ul>
</li>
<li class="nav-item" data-level="2"><a class="nav-link" href="#installation">INSTALLATION</a>
<ul class="nav flex-column">
</ul>
</li>
<li class="nav-item" data-level="2"><a class="nav-link" href="#usage">USAGE</a>
<ul class="nav flex-column">
</ul>
</li>
<li class="nav-item" data-level="2"><a class="nav-link" href="#input">Input</a>
<ul class="nav flex-column">
</ul>
</li>
<li class="nav-item" data-level="2"><a class="nav-link" href="#output">Output</a>
<ul class="nav flex-column">
</ul>
</li>
<li class="nav-item" data-level="2"><a class="nav-link" href="#annotation">Annotation</a>
<ul class="nav flex-column">
</ul>
</li>
<li class="nav-item" data-level="2"><a class="nav-link" href="#example">Example</a>
<ul class="nav flex-column">
</ul>
</li>
<li class="nav-item" data-level="2"><a class="nav-link" href="#weight-generation">Weight generation</a>
<ul class="nav flex-column">
</ul>
</li>
<li class="nav-item" data-level="2"><a class="nav-link" href="#acknowledgements">ACKNOWLEDGEMENTS</a>
<ul class="nav flex-column">
</ul>
</li>
<li class="nav-item" data-level="2"><a class="nav-link" href="#appendix">APPENDIX</a>
<ul class="nav flex-column">
</ul>
</li>
<li class="nav-item" data-level="2"><a class="nav-link" href="#bioconductor-packages">Bioconductor packages</a>
<ul class="nav flex-column">
</ul>
</li>
<li class="nav-item" data-level="2"><a class="nav-link" href="#references">REFERENCES</a>
<ul class="nav flex-column">
</ul>
</li>
</ul>
</div>
</div></div>
<div class="col-md-9" role="main">
<h2 id="introduction">INTRODUCTION</h2>
<p>The title represents pipeline for Epigenomewide association study (EWAS) and Functional Summary-based Imputation (FUSION) association analyses.</p>
<p>Transcriptomewide association statistic <span class="arithmatex">\(z_{TWAS}\)</span> was originally proposed for gene expression data. For a given Trait of interest <strong>T</strong> for which GWAS summary statistics <span class="arithmatex">\(z_T\)</span> is available, the corresponding Wald statistic for TWAS is defined such that
$$
z_{TWAS} = \frac{w^T_{ge}z_T}{\sqrt{w^T_{ge}Vw_{ge}}}
$$</p>
<p>where <span class="arithmatex">\(w_{ge}\)</span> is a weight associated with gene expression and <strong>V</strong> covariance matrix for <span class="arithmatex">\(z_T\)</span>, respectively.</p>
<p>By analogy, an epigenomewide association statistic <span class="arithmatex">\(z_{EWAS}\)</span> is defined through methylation data so that
$$
z_{EWAS} = \frac{w^T_{me}z_T}{\sqrt{w^T_{me}Vw_{me}}}
$$</p>
<p>where <span class="arithmatex">\(w_{me}\)</span> is the weight associated with methylation. Both approaches allow for imputation using GWAS summary statistics. The derivation of these weights and imputation were done using methods as described in @gusev16 called TWAS as well as in @mancuso17 called Functional Summary-based Imputation (FUSION). The TWAS statistics from both approaches agreed very well.</p>
<div class="mermaid">graph TB;
SNP["LD reference panel (bed,bim,fam)"] --&gt; |"EWAS reference panel(top1, blup, lasso, enet, bslmm)"| Methylation;
Methylation --&gt; Trait;
SNP --&gt; |"GWAS summary statistics (SNP, A1, A2, Z)"| Trait;
</div>
<p>A total of 442,920 CpG sites based on Illumina humanmethylation450 chips on 1,.146 individuals in EPIC-Norfolk study were available. Among these, 1,117 individuals also had genotype data from Affymetrix BioBank Axiom chips. HapMap2 SNPs from genetic data of these individuals were extracted via PLINK2 according to cis-positions of each probe and subsequently used to build weight analogous to gene expression data as implemented in computer software TWAS. We filtered probes according to their heritabilities estimated from software GCTA at significant level of 0.01. We then performed EWAS for given GWAS summary statistics. The weight generation and methylation imputation was implemented in software called TWAS-pipeline, which allows for whole epigenome computation. After filtering, 78,133 probes reached significant level 0.01.</p>
<p>The FUSION framework has several advantages: First, it integrates heritability estimation and covariate adjustment for whole-chromosomes with additional models such as LASSO, elastic net, BLUP. Second, it offers cross-validation, joint/conditional analyses with the output also informing top hit SNPs and inferred methylation quantitative trait locus (meQTL). Besides, the new software uses modified GCTA software (gcta_nr_robust) leading to higher yield of probes with heritabilities reaching statistical significance, GEMMA giving BSLMM estimates and ability to align strands with reference panels. As both the increased number of models and cross-validation led to excessive computing time, we dropped BSLMM models and conducted five cross-validations. As a result our reference panel for EWAS imputation contains 77,372 probes reaching the heritability p value threshold of 0.01. The association as well as joint/conditional analysis using our weights and LD panel is implemented in software called EWAS-fusion. Like the original TWAS, our implementation will enable a range of GWAS summary statistics to be used coupled with downstream analysis.</p>
<p>EWAS-fusion is reminiscent of Mendelian Randomisation as shown below,</p>
<div class="mermaid">graph TB;
SNP --&gt; |"EWAS reference panel(top1, blup, lasso, enet, bslmm)"| Methylation;
Methylation --&gt; Trait;
SNP --&gt; |"GWAS summary statistics (SNP, A1, A2, Z)"| Trait;
</div>
<h2 id="installation">INSTALLATION</h2>
<ul>
<li>
<p>To begin, the software <a href="http://gusevlab.org/projects/fusion/">FUSION</a> including dependencies such as <code>plink2R</code> and <code>reshape</code> is required. The latest version also requires <a href="https://github.com/cotsapaslab/jlim">jlimR</a>. Other facilities to be required are Sun grid engine (sge) or GNU parallel for Linux clusters. A summary is also available from here: <a href="https://jinghuazhao.github.io/software-notes/AA/">https://jinghuazhao.github.io/software-notes/AA/</a>.</p>
</li>
<li>
<p>Install the repository on your system, you will need weights based on epigenetic data or to generate them as described in <strong>Weight generation</strong> below.</p>
</li>
</ul>
<table>
<thead>
<tr>
<th>FILE</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>EWAS-weights/</td>
<td>directory for EWAS weights</td>
</tr>
<tr>
<td>glist-hg19</td>
<td>Probe list</td>
</tr>
<tr>
<td>LDREF/</td>
<td>Reference for LD</td>
</tr>
<tr>
<td>EWAS-weights.pos</td>
<td>Definition of regions</td>
</tr>
<tr>
<td>EWAS-weights.profile*</td>
<td>Probe profiles</td>
</tr>
</tbody>
</table>
<p><code>*</code> It contains information about the probes but not directly involved in the association analysis. Earlier version of EWAS-fusion used EWAS/, RDat.pos, and RDat.profile.</p>
<h2 id="usage">USAGE</h2>
<p>The syntax is as follows,</p>
<pre><code>ewas-fusion.sh input-file
</code></pre>
<p>These will send jobs to the Linux clusters. The sge error and output, if any, should be called EWAS.e and EWAS.o in your HOME directory.</p>
<h2 id="input">Input</h2>
<p>The input file contains GWAS summary statistics similar to <a href="https://github.com/bulik/ldsc/wiki/Summary-Statistics-File-Format">.sumstats</a> as in <a href="https://github.com/bulik/ldsc">LDSC</a> with the following columns.</p>
<table>
<thead>
<tr>
<th>Column</th>
<th>Name</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>1</td>
<td>SNP</td>
<td>RS id of SNPs</td>
</tr>
<tr>
<td>2</td>
<td>A1</td>
<td>Effect allele (first allele)</td>
</tr>
<tr>
<td>3</td>
<td>A2</td>
<td>Other allele (second allele)</td>
</tr>
<tr>
<td>4</td>
<td>Z</td>
<td>Z-scores, taking sign with repect to A1</td>
</tr>
</tbody>
</table>
<h2 id="output">Output</h2>
<p>The results will be in input-file.tmp/ directory.</p>
<h2 id="annotation">Annotation</h2>
<p>This is furnished with contribution from Dr Alexia Cardona, alexia.cardona@mrc-epid.cam.ac.uk, as follows,</p>
<pre><code>Rscript ewas-annotate.R input-file.tmp
</code></pre>
<p>It is assumed that <code>HumanMethylation450_15017482_v1-2.csv</code> is available from the directory containing <code>ewas-annotate.R</code>but this can be at different location</p>
<pre><code>Rscript ewas-annotate.R input-file.tmp manifest_location=/at/different/location
</code></pre>
<p>Q-Q and Manhattan plots using R/gap can be obtained from</p>
<pre><code>Rscript ewas-plot.R input-file.tmp
</code></pre>
<h2 id="example">Example</h2>
<p>The script <a href="test.sh">test.sh</a> uses data reported in @wood14. It downloads and generates an input file called <code>height</code> to <code>ewas-fusion.sh</code>. </p>
<pre><code>ewas-fusion.sh height
</code></pre>
<p>The results will be in <code>height.tmp/</code> once it is done.</p>
<p>The annotation is done with</p>
<pre><code>Rscript ewas-annotate.R height.tmp
</code></pre>
<p>The Q-Q and Manhattan plots are generated with</p>
<pre><code>Rscript ewas-plot.R height.tmp
</code></pre>
<h2 id="weight-generation">Weight generation</h2>
<p>This is a revised and much simplified implementation of codes available from TWAS-pipeline. Under our sge it is furnished with</p>
<pre><code>qsub get_weight.qsub
</code></pre>
<p>or</p>
<pre><code>qsub get_weight.qsub 22
</code></pre>
<p>for chromosome 22.</p>
<p>Inputs to these are summarised as follows, </p>
<table>
<thead>
<tr>
<th>File</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>FUSION.pheno</td>
<td>PLINK phenotype file containing data for all probes</td>
</tr>
<tr>
<td>FUSION.covar</td>
<td>PLINK covariate file containing covariates such as PCs</td>
</tr>
<tr>
<td>CpG.txt</td>
<td>CpG ID, chromosome and position</td>
</tr>
</tbody>
</table>
<p>In addition, PLINK binary pedigree file for each CpG also requires to be prepared, as in <a href="files">files</a>. Although it was not done, it is possible to use code as in <a href="files/1KG.sh">1KG.sh</a> to get around gerneration of these individual files by using a combined one. Note the setup takes advantage of the compact storage of non-genetic data.</p>
<p>The results will be available from the EWAS-fusion directory to be profiled and used for association analysis above. As the number of files is fairly large, <a href="files/cp_weight.qsub">cp_weight.qsub</a> is written to put weights from their temporary directories in place while <a href="files/ewas-profile.sh">ewas-profile.sh</a> profiles these weights as well as prepares for LDREF. For <a href="https://data.broadinstitute.org/alkesgroup/FUSION/LDREF.tar.bz2">the version with FUSION</a>, it can be done as follows,</p>
<pre><code class="language-bash">wget -qO- https://data.broadinstitute.org/alkesgroup/FUSION/LDREF.tar.bz2 | tar xfj - --strip-components=1
seq 22|awk -vp=1000G.EUR. '{print p $1}' &gt; merge-list
plink-1.9 --merge-list merge-list --make-bed --out FUSION
rm 1000G.EUR.* merge-list
sort -k2,2 FUSION.bim &gt; EUR.bim
</code></pre>
<p>To mirror FUSION, which uses <a href="https://www.cog-genomics.org/static/bin/plink/glist-hg19">glist-hg19</a>, an equivalent for EWAS needs to be built.</p>
<h2 id="acknowledgements">ACKNOWLEDGEMENTS</h2>
<p>We wish to thank colleagues and collaborators for their invaluable contributions to make this work possible.</p>
<h2 id="appendix">APPENDIX</h2>
<p>Additional information for Illumina infinium humanmethylation450 beadchip as in <a href="https://support.illumina.com/array/array_kits/infinium_humanmethylation450_beadchip_kit/downloads.html">Illumina website</a></p>
<table>
<thead>
<tr>
<th><strong>Column Name</strong></th>
<th><strong>Description</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>Index</td>
<td>Probe Index</td>
</tr>
<tr>
<td>TargetID</td>
<td>Identifies the probe name. Also used as a key column for data import.</td>
</tr>
<tr>
<td>ProbeID_A</td>
<td>Illumina identifier for probe sequence A</td>
</tr>
<tr>
<td>ProbeID_B</td>
<td>Illumina identifier for probe sequence B</td>
</tr>
<tr>
<td>IlmnID</td>
<td>Unique CpG locus identifier from the Illumina CG database</td>
</tr>
<tr>
<td>Name</td>
<td>Unique CpG locus identifier from the Illumina CG database</td>
</tr>
<tr>
<td>AddressA_ID</td>
<td>Address of probe A</td>
</tr>
<tr>
<td>AlleleA_ProbeSeq</td>
<td>Sequence for probe A</td>
</tr>
<tr>
<td>AddressB_ID</td>
<td>Address of probe  B</td>
</tr>
<tr>
<td>AlleleB_ProbeSeq</td>
<td>Sequence for probe B</td>
</tr>
<tr>
<td>Infinium_Design_Type</td>
<td>Defines Assay type - Infinium I or Infinium II</td>
</tr>
<tr>
<td>Next_Base</td>
<td>Base added at SBE step - Infinium I assays only</td>
</tr>
<tr>
<td>Color_Channel</td>
<td>Color of the incorporated baseá (Red or Green) - Infinium I assays only</td>
</tr>
<tr>
<td>Forward_Sequence</td>
<td>Sequence (in 5'-3' orientation) flanking query site</td>
</tr>
<tr>
<td>Genome_Build</td>
<td>Genome build on which forward sequence is based</td>
</tr>
<tr>
<td>CHR</td>
<td>Chromosome - genome build 37</td>
</tr>
<tr>
<td>MAPINFO</td>
<td>Coordinates - genome build 37</td>
</tr>
<tr>
<td>SourceSeq</td>
<td>Unconverted design sequence</td>
</tr>
<tr>
<td>Chromosome_36</td>
<td>Chromosome - genome build 36</td>
</tr>
<tr>
<td>Coordinate_36</td>
<td>Coordinates - genome build 36</td>
</tr>
<tr>
<td>Strand</td>
<td>Design strand</td>
</tr>
<tr>
<td>Probe_SNPs</td>
<td>Assays with SNPs present within probe &gt;10bp from query site</td>
</tr>
<tr>
<td>Probe_SNPs_10</td>
<td>Assays with SNPs present within probe ?10bp from query site (HM27 carryover or recently discovered)</td>
</tr>
<tr>
<td>Random_Loci</td>
<td>Loci which were chosen randomly in the design proccess</td>
</tr>
<tr>
<td>Methyl27_Loci</td>
<td>Present or absent on HumanMethylation27 array</td>
</tr>
<tr>
<td>UCSC_RefGene_Name</td>
<td>Gene name (UCSC)</td>
</tr>
<tr>
<td>UCSC_RefGene_Accession</td>
<td>Accession number (UCSC)</td>
</tr>
<tr>
<td>UCSC_RefGene_Group</td>
<td>Gene region feature category (UCSC)</td>
</tr>
<tr>
<td>UCSC_CpG_Islands_Name</td>
<td>CpG island name (UCSC)</td>
</tr>
<tr>
<td>Relation_to_UCSC_CpG_Island</td>
<td>Relationship to Canonical CpG Island: Shores - 0-2 kb from CpG island; Shelves - 2-4 kb from CpG island.</td>
</tr>
<tr>
<td>Phantom</td>
<td>FANTOM-derived promoter</td>
</tr>
<tr>
<td>DMR</td>
<td>Differentially methylated region (experimentally determined)</td>
</tr>
<tr>
<td>Enhancer</td>
<td>Enhancer element (informatically-determined)</td>
</tr>
<tr>
<td>HMM_Island</td>
<td>Hidden Markov Model Island</td>
</tr>
<tr>
<td>Regulatory_Feature_Name</td>
<td>Regulatory feature (informatically determined)</td>
</tr>
<tr>
<td>Regulatory_Feature_Group</td>
<td>Regulatory feature category</td>
</tr>
<tr>
<td>DHS</td>
<td>DNAse hypersensitive site (experimentally determined)</td>
</tr>
</tbody>
</table>
<h2 id="bioconductor-packages">Bioconductor packages</h2>
<p>These are <strong>IlluminaHumanMethylation450kanno.ilmn12.hg19</strong> and <strong>IlluminaHumanMethylation450kmanifest</strong> as shown in <strong>minfiDataEPIC</strong>.</p>
<pre><code class="language-r">library(IlluminaHumanMethylation450kanno.ilmn12.hg19)
data(IlluminaHumanMethylation450kanno.ilmn12.hg19)
options(width=200)
annotation.table &lt;- getAnnotation(IlluminaHumanMethylation450kanno.ilmn12.hg19)
head(Locations)
head(Manifest)
head(annotation.table)
</code></pre>
<p>with <code>IlluminaHumanMethylation450kanno.ilmn12.hg19</code> as follows,</p>
<pre><code>IlluminaMethylationAnnotation object
Annotation
  array: IlluminaHumanMethylation450k
  annotation: ilmn12
  genomeBuild: hg19
Available annotation
  Islands.UCSC
  Locations
  Manifest
  Other
  SNPs.132CommonSingle
  SNPs.135CommonSingle
  SNPs.137CommonSingle
  SNPs.138CommonSingle
  SNPs.141CommonSingle
  SNPs.142CommonSingle
  SNPs.144CommonSingle
  SNPs.146CommonSingle
  SNPs.147CommonSingle
  SNPs.Illumina
Defaults
  Locations
  Manifest
  SNPs.137CommonSingle
  Islands.UCSC
  Other
</code></pre>
<h2 id="references">REFERENCES</h2>
<p>Freund MK, Burch KS, Shi H, Mancuso N, Kichaev G, Garske KM, Pan DZ, Miao Z, Mohlke KL, Laakso M, Pajukanta P, Pasaniuc B, Arboleda VA. 2018. Phenotype-specific enrichment of mendelian disorder genes near GWAS regions across 62 complex traits. Am J Hum Genet 103: 535–552.</p>
<p>Gusev A, Ko A, Shi H, Bhatia G, Chung W, Penninx BW, Jansen R, Geus EJ de, Boomsma DI, Wright FA, Sullivan PF, Nikkola E, Alvarez M, Civelek M, Lusis AJ, Lehtimäki T, Raitoharju E, Kähönen M, Seppälä I, Raitakari OT, Kuusisto J, Laakso M, Price AL, Pajukanta P, Pasaniuc B. 2016. Integrative approaches for large-scale transcriptome-wide association studies. Nat Genet 48: 245–52.</p>
<p>Mancuso N, Shi H, Goddard P, Kichaev G, Gusev A, Pasaniuc B. 2017. Integrating gene expression with summary association statistics to identify genes associated with 30 complex traits. Am J Hum Genet 100: 473–487.</p>
<p>Raj T, Li YI, Wong G, Humphrey J, Wang M, Ramdhani S, Wang YC, Ng B, Gupta I, Haroutunian V, Schadt EE, Young-Pearse T, Mostafavi S, Zhang B, Sklar P, Bennett DA, De Jager PL. 2018. Integrative transcriptome analyses of the aging brain implicate altered splicing in alzheimer’s disease susceptibility. Nat Genet 50: 1584–1592.</p>
<p>Turner SD. 2018. Qqman: An r package for visualizing GWAS results using q-q and manhattan plots. Journal of Open Source Software 3: 731.</p>
<p>Wood AR, Esko T, Yang J, Vedantam S, Pers TH, Gustafsson S, Chu AY, Estrada K, Luan J, Kutalik Z, Amin N, Buchkovich ML, Croteau-Chonka DC, Day FR, Duan Y, Fall T, Fehrmann R, Ferreira T, Jackson AU, Karjalainen J, Lo KS, Locke AE, Mägi R, Mihailov E, Porcu E, Randall JC, Scherag A, Vinkhuyzen AA, Westra HJ, Winkler TW, Workalemahu T, Zhao JH, Absher D, Albrecht E, Anderson D, Baron J, Beekman M, Demirkan A, Ehret GB, Feenstra B, Feitosa MF, Fischer K, Fraser RM, Goel A, Gong J, Justice AE, Kanoni S, Kleber ME, Kristiansson K, Lim U, Lotay V, Lui JC, Mangino M, Mateo Leach I, Medina-Gomez C, Nalls MA, Nyholt DR, Palmer CD, Pasko D, Pechlivanis S, Prokopenko I, Ried JS, Ripke S, Shungin D, Stancáková A, Strawbridge RJ, Sung YJ, Tanaka T, Teumer A, Trompet S, Laan SW van der, Setten J van, Van Vliet-Ostaptchouk JV, Wang Z, Yengo L, Zhang W, Afzal U, Arnlöv J, Arscott GM, Bandinelli S, Barrett A, Bellis C, Bennett AJ, Berne C, Blüher M, Bolton JL, Böttcher Y, Boyd HA, Bruinenberg M, Buckley BM, Buyske S, Caspersen IH, Chines PS, Clarke R, Claudi-Boehm S, Cooper M, Daw EW, De Jong PA, et al. 2014. Defining the role of common variation in the genomic and biological architecture of adult human height. Nat Genet 46: 1173–86.</p>
<p>Zhao JH. 2007. <strong>gap</strong>: genetic analysis package. 2007 23: 18.</p>
<p>Zhao JH. 2021. Weights of EWAS-fusion. <a href="https://doi.org/10.5281/zenodo.5069993">https://doi.org/10.5281/zenodo.5069993</a>.</p></div>
</div>
</div>
<footer class="col-md-12">
<hr/>
<p>Documentation built with <a href="https://www.mkdocs.org/">MkDocs</a>.</p>
</footer>
<script>
            var base_url = ".",
                shortcuts = {"help": 191, "next": 78, "previous": 80, "search": 83};
        </script>
<script defer="" src="js/base.js"></script>
<script defer="" src="javascripts/config.js"></script>
<script defer="" src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
<script defer="" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
<script defer="" src="search/main.js"></script>
<div aria-hidden="true" aria-labelledby="searchModalLabel" class="modal" id="mkdocs_search_modal" role="dialog" tabindex="-1">
<div class="modal-dialog modal-lg">
<div class="modal-content">
<div class="modal-header">
<h4 class="modal-title" id="searchModalLabel">Search</h4>
<button class="close" data-dismiss="modal" type="button"><span aria-hidden="true">×</span><span class="sr-only">Close</span></button>
</div>
<div class="modal-body">
<p>
                    From here you can search these documents. Enter
                    your search terms below.
                </p>
<form>
<div class="form-group">
<input class="form-control" id="mkdocs-search-query" placeholder="Search..." title="Type search term here" type="search"/>
</div>
</form>
<div id="mkdocs-search-results"></div>
</div>
<div class="modal-footer">
</div>
</div>
</div>
</div><div aria-hidden="true" aria-labelledby="keyboardModalLabel" class="modal" id="mkdocs_keyboard_modal" role="dialog" tabindex="-1">
<div class="modal-dialog">
<div class="modal-content">
<div class="modal-header">
<h4 class="modal-title" id="keyboardModalLabel">Keyboard Shortcuts</h4>
<button class="close" data-dismiss="modal" type="button"><span aria-hidden="true">×</span><span class="sr-only">Close</span></button>
</div>
<div class="modal-body">
<table class="table">
<thead>
<tr>
<th style="width: 20%;">Keys</th>
<th>Action</th>
</tr>
</thead>
<tbody>
<tr>
<td class="help shortcut"><kbd>?</kbd></td>
<td>Open this help</td>
</tr>
<tr>
<td class="next shortcut"><kbd>n</kbd></td>
<td>Next page</td>
</tr>
<tr>
<td class="prev shortcut"><kbd>p</kbd></td>
<td>Previous page</td>
</tr>
<tr>
<td class="search shortcut"><kbd>s</kbd></td>
<td>Search</td>
</tr>
</tbody>
</table>
</div>
<div class="modal-footer">
</div>
</div>
</div>
</div>
<script src="https://unpkg.com/mermaid@8.8.0/dist/mermaid.min.js"></script><script>mermaid.initialize({});</script></body>
</html>
<!--
MkDocs version : 1.1.2
Build Date UTC : 2021-08-24 20:01:36.175776+00:00
-->
